Composer/packages/ui-plugins/composer/src/defaultFlowSchema.ts

```diff
@@ -0,0 +1,195 @@
+  [SDKKinds.SendActivity]: {
+    widget: 'ActionCard',
+    body: '=action.activity',
+  },
+  [SDKKinds.AttachmentInput]: {
+    widget: 'ActionCard',
+    body: '=action.prompt',
+  },
+  [SDKKinds.ConfirmInput]: {
+    widget: 'ActionCard',
+    body: '=action.prompt',
+  },
+  [SDKKinds.DateTimeInput]: {
+    widget: 'ActionCard',
+    body: '=action.prompt',
+  },
+  [SDKKinds.NumberInput]: {
+    widget: 'ActionCard',
+    body: '=action.prompt',
+  },
+  [SDKKinds.TextInput]: {
+    widget: 'ActionCard',
+    body: '=action.prompt',
+  },
+  [SDKKinds.ChoiceInput]: {
+    widget: 'ActionCard',
+    body: '=action.prompt',
+  },
+  [SDKKinds.BeginDialog]: {
+    widget: 'ActionCard',
+    body: {
+      widget: 'DialogRef',
+      dialog: '=action.dialog',
+    },
+    footer: {
+      widget: 'PropertyDescription',
+      property: '=action.resultProperty',
+      description: '=string("= Return value")',
+    },
+    hideFooter: '=!action.resultProperty',
+  },
+  [SDKKinds.BeginSkill]: {
+    widget: 'ActionCard',
+    colors: { theme: '#004578', color: '#FFFFFF', icon: '#FFFFFF' },
+    icon: 'Library',
+    body: {
+      widget: 'ResourceOperation',
+      operation: 'Host',
+      resource: '=coalesce(action.skillEndpoint, "?")',
+      singleline: true,
+    },
+    footer: {
+      widget: 'PropertyDescription',
+      property: '=action.resultProperty',
+      description: '=string("= Result")',
+    },
+    hideFooter: '!action.resultProperty',
+  },
+  [SDKKinds.ReplaceDialog]: {
+    widget: 'ActionCard',
+    body: {
+      widget: 'DialogRef',
+      dialog: '=action.dialog',
+    },
+  },
+  [SDKKinds.EditArray]: {
+    widget: 'ActionCard',
+    body: {
+      widget: 'ResourceOperation',
+      operation: '=coalesce(action.changeType, "?")',
+      resource: '=coalesce(action.itemsProperty, "?")',
+    },
+    footer: {
+      widget: 'PropertyDescription',
+      property: '=action.resultProperty',
+      description: '= Result',
+    },
+    hideFooter: '=!action.resultProperty',
+  },
+  [SDKKinds.SetProperty]: {
+    widget: 'ActionCard',
+    body: '=concat(coalesce(action.property, "?"), " : ", coalesce(action.value, "?"))',
+  },
+  [SDKKinds.SetProperties]: {
+    widget: 'ActionCard',
+    body: {
+      widget: 'ListOverview',
+      items: '=foreach(action.assignments, x => concat(coalesce(x.property, "?"), " : ", coalesce(x.value, "?")))',
+    },
+  },
+  [SDKKinds.DeleteProperty]: {
+    widget: 'ActionCard',
+    body: '=action.property',
+  },
+  [SDKKinds.DeleteProperties]: {
+    widget: 'ActionCard',
+    body: {
+      widget: 'ListOverview',
+      items: '=action.properties',
+    },
+  },
+  [SDKKinds.DeleteActivity]: {
+    widget: 'ActionCard',
+    header: {
+      widget: 'ActionHeader',
+      title: 'Delete activity',
+    },
+    body: {
+      widget: 'PropertyDescription',
+      property: '=coalesce(action.activityId, "?")',
+      description: '=string("= ActivityId")',
```

Comment:
Is this really necessary? Couldn't this just be a string?

Reply:
It begins with '=', causes the expression engine will recognize it as a valid expression and returns `undefined` while `= Result property` won't.

To escapt this case, either using `=concat("= ActivityId")` or using `=string()`

Reply:
So this is a special case where we want the string to start with a `=` but that is seen as an expression? Can we escape it? `"\= ActivityId"`? Or attempt to parse as an expression then fallback to this string if that fails?

Having to write an expression for this isn't very nice.

Reply:
`"\= ActivityId"` won't be recognized as expression but will be displayed as it is with a prefix `\`, so seems we can't write it that way.
Another tricky point here is, `= Activity` can be evaluated successfully by Exp while `= Activity property` throws an error and can fallback, so as a result, we don't need to wrap  `"= Result property"` but we have to wrap `"= ActivityId"`.

I think the `"="` problem is the cost of introducing Expression engine. There will always be an indicater to mark a plain string as Expression, either the `"="` or other characters. Under current expression engine's capability, the most concise way to write that is "=\`= Activity\`"

Reply:
Another possible solution here is we detect the expression pattern with Regex, only when an expression contains `action.xxx` pattern do we evaluate it with Expression.
The assumption is, in the Flow schema, an meaningful expression have to read action data by writing `=...action.xxx`  .

```ts
if (exp.beginWith("=") && exp.indexOf("action.") > -1) {
  return Expression.evaluate(exp, context);
}
```

Reply:
resolved, '=string()' no longer needed